qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/AppointmentSteps.java

```diff
@@ -0,0 +1,241 @@
+import org.openmrs.contrib.qaframework.page.ManageAppointmentsPage;
+import org.openmrs.contrib.qaframework.page.ManageProviderSchedulesPage;
+import org.openmrs.contrib.qaframework.page.RequestAppointmentPage;
+
+public class AppointmentSteps extends Steps {
+
+	private AppointmentSchedulingPage appointmentSchedulingPage;
+	private AppointmentBlocksPage appointmentBlocksPage;
+	private ManageAppointmentsPage manageAppointmentsPage;
+	private ManageProviderSchedulesPage manageProviderSchedulesPage;
+	private RequestAppointmentPage requestAppointmentPage;
+	private TestData.PatientInfo testPatient;
+	private static final String FIRST_SERVICE_NAME = "Oncology";
+	private static final String SECOND_SERVICE_NAME = "Dermatology";
+	private static final String LOCATION = "Laboratory";
+
+	@Before(RunTest.HOOK.SELENIUM_APPOINTMENT)
+	public void visitDashboard() {
+		initiateWithLogin();
+		testPatient = createTestPatient();
+		findPatientPage = homePage.goToFindPatientRecord();
+		findPatientPage.enterPatient(testPatient.identifier);
+		findPatientPage.waitForPageToLoad();
+		dashboardPage = findPatientPage.clickOnFirstPatient();
+	}
+
+	@After(RunTest.HOOK.SELENIUM_APPOINTMENT)
+	public void destroy() {
+		deletePatient(testPatient);
+		quit();
+	}
+
+	@Given("a user clicks on Request Appointment link from the Patient dashboard")
+	public void loadRequestAppointmentPage() {
+		requestAppointmentPage = (RequestAppointmentPage) dashboardPage.clickOnRequest();
+	}
+
+	@When("the system loads Request Appointment page")
+	public void systemLoadsRequestAppointmentPage() {
+		assertTrue(textExists("Request Appointment"));
+	}
+
+	@And("a user fills in the appointment request details")
+	public void captureAppointmentRequestDetails() {
+		requestAppointmentPage.enterAppointmentType(FIRST_SERVICE_NAME);
+		requestAppointmentPage.enterMinimumValue("0");
+		requestAppointmentPage.selectMinimumUnits("Day(s)");
+		requestAppointmentPage.enterMaximumValue("2");
+		requestAppointmentPage.selectMaximumUnits("Day(s)");
+		requestAppointmentPage.addAppointmentRequestNote("I need to see a doctor first");
+	}
+
+	@And("a user clicks save button")
+	public void saveAppointmentRequest() {
+		dashboardPage = requestAppointmentPage.saveRequest();
+	}
+
+	@Then("the system adds appointment request into the appointment requests table")
+	public void systemSavesAppointmentRequest() {
+		assertNotNull(dashboardPage.getAppointmentRequestsList());
+	}
+
+	@When("a user clicks on Schedule Appointment link from the Patient dashboard")
+	public void loadManageAppointmentsPage() {
+		manageAppointmentsPage = (ManageAppointmentsPage) dashboardPage.goToManageAppointments();
+	}
+
+	@And("the system loads Manage Appointments page")
+	public void systemLoadsManageAppointmentsPage() {
+		assertTrue(manageAppointmentsPage.containsText(testPatient.identifier));
+	}
+
+	@And("a user clicks on Cancel this Request icon")
+	public void deleteAppointmentRequest() {
```

Comment:
Same here
```suggestion
	public void userClicksOnDeleteAppointmentRequest() {
```

Reply:
> Same here

@sherrif10 I would still think that deleteAppointmentRequest method would suffice. Using userClicksOnDeleteAppointmentRequest method to me would seem like there's another step the user needs to perform for deletion to be effected.

Reply:
@sherrif10 do you remember advising me to have steps that effect an operation so that we do not have so many other minor steps required. That is the view point I am standing on.

Reply:
> Using userClicksOnDeleteAppointmentRequest method to me would seem like there's another step the user needs to perform for deletion to be effected.

Not like that, you are clicking the button, or the user clicking the button, so i think userClick is betterthan simply deleting

Reply:
> > Using userClicksOnDeleteAppointmentRequest method to me would seem like there's another step the user needs to perform for deletion to be effected.
> 
> Not like that, you are clicking the button, or the user is clicking the button, so i think **userClick** is betterthan simply **deleting**

Ok, if you still say so then I have to change it @sherrif10

Reply:
Yeah it does

Reply:
@sherrif10 I have attended to the reviews and pushed  a commit